### PR TITLE
Added GovernorVotesCompQuorumFraction for settable quorum

### PR DESCRIPTION
**Pull request by jferas**
Added GovernorVotesCompQuorumFraction for settable quorum

Also added test for setting quorum numerator

### CODE REVIEW

**Comment by jferas** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/ScopeLift/radworks-governor-upgrade/1?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/ScopeLift/radworks-governor-upgrade/2?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5** <a href="https://app.graphite.dev/github/pr/ScopeLift/radworks-governor-upgrade/5?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #8** <a href="https://app.graphite.dev/github/pr/ScopeLift/radworks-governor-upgrade/8?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #12** <a href="https://app.graphite.dev/github/pr/ScopeLift/radworks-governor-upgrade/12?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #13** <a href="https://app.graphite.dev/github/pr/ScopeLift/radworks-governor-upgrade/13?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/radworks-governor-upgrade/13?utm_source=stack-comment).

**Comment by jferas** [comment]
Subsequent commit on this PR implements a more code efficient settable quorum.

File: src/RadworksGovernor.sol [hunk 1]
```diff
@@ -177,8 +175,13 @@ contract RadworksGovernor is
   /// @notice The amount of RAD required to meet the quorum threshold for a proposal
   /// as of a given block.
   /// @dev Our implementation ignores the block number parameter and returns a constant.
-  function quorum(uint256) public pure override returns (uint256) {
-    return QUORUM;
+  function quorum(uint256 timepoint)
```
Comment (alexkeating) [inline src/RadworksGovernor.sol:178]: (arbitrary comment)

Have we decided which implementation we wanted to use?

File: src/lib/GovernorVotesCompQuorumFraction.sol [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+} from "@openzeppelin/contracts/governance/extensions/GovernorVotesComp.sol";
+import "@openzeppelin/contracts/utils/Checkpoints.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/**
+ * @dev Extension of {Governor} for voting weight extraction from an {GovernorVotesComp} token and a
+ * quorum
+ * expressed as a fraction of the total supply.
+ */
+abstract contract GovernorVotesCompQuorumFraction is GovernorVotesComp {
```
[the remark this answers is not in the document]
Comment (jferas) [reply src/lib/GovernorVotesCompQuorumFraction.sol:19]: Done: 34b7f0174bbde5827eecc7f2b69fe9b066795826

File: src/lib/GovernorVotesCompQuorumFraction.sol [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+   * Emits a {QuorumNumeratorUpdated} event.
+   *
+   * Requirements:
+   *
+   * - New numerator must be smaller or equal to the denominator.
+   */
+  function _updateQuorumNumerator(uint256 newQuorumNumerator) internal virtual {
+    require(
+      newQuorumNumerator <= quorumDenominator(),
+      "GovernorVotesQuorumFraction: quorumNumerator over quorumDenominator"
```
[the remark this answers is not in the document]
Comment (jferas) [reply src/lib/GovernorVotesCompQuorumFraction.sol]: Done: 34b7f0174bbde5827eecc7f2b69fe9b066795826

File: src/lib/GovernorVotesCompQuorumFraction.sol [hunk 4]
```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+// (Based on OpenZeppelin governance/extensions/GovernorVotesQuorumFraction.sol)
```
[the remark this answers is not in the document]
Comment (jferas) [reply src/lib/GovernorVotesCompQuorumFraction.sol]: Done: 34b7f0174bbde5827eecc7f2b69fe9b066795826

**Comment by jferas** [comment]
Radworks has chose to go with a fixed, constant quorum.

**Pull request closed without merging**